src/robotFactory.js

```diff
@@ -1,17 +1,73 @@
 'use strict';
 
-function BaseRobot() {
-  // implement
+function BaseRobot(name, weight, coords, chipVersion) {
+  this.name = name;
+  this.weight = weight;
+  this.coords = coords;
+  this.chipVersion = chipVersion;
 }
 
-function FlyingRobot() {
-  // implement
-}
+BaseRobot.prototype.goForward = function(step = 1) {
+  this.coords.y += step;
+};
+
+BaseRobot.prototype.goBack = function(step = 1) {
+  this.coords.y -= step;
+};
```

Comment:
```suggestion
BaseRobot.prototype = {
  goForward(step = 1) {
    this.coords.y += step;
  },
  goBack(step = 1) {
    this.coords.y -= step;
  },
};
```
используй подобный метод

Reply:
@Vladimir-Beliy в остальных местах мы пишем FlyingRobot.prototype = BaseRobot.prototype; 
получается, он уже есть, и мы к нему новые свойства добавляем. разве не будет ошибкой если я заново напишу FlyingRobot.prototype = {}?